lib/index.js

```diff
@@ -120,46 +120,74 @@ Optimizely.prototype.page = function(page) {
  * @param {String} experimentState.variations[].name: the name of the variation
  * @param {String} experimentState.variations[].id: the ID of the variation
- * @param {Object} experimentState.section: the section for the variations (only defined for multivariate experiments)
- * @param {Object} experimentState.section.name: the name of section
- * @param {Object} experimentState.section: the ID of the section
+ * @param {Object} experimentState.sections: the sections for the experiment (only defined for multivariate experiments) keyed by sectionId
+ * @param {Object} experimentState.sections[sectionId].name: the name of section
+ * @param {Object} experimentState.sections[sectionId].variation_ids: the IDs of the variations in the section
  *
  */
 
 Optimizely.prototype.sendClassicDataToSegment = function(experimentState) {
   var experiment = experimentState.experiment;
   var variations = experimentState.variations;
-  var section = experimentState.section;
+  var sections = experimentState.sections;
   var variationNames = [];
+  var variationIds = [];
 
   // Makes concatenating easier later
   each(function(variation) {
     variationNames.push(variation.name);
+    variationIds.push(variation.id);
   }, variations);
 
   // Send data via `.track()`
   if (this.options.listen) {
     var props = {
       experimentId: experiment.id, // to be consistent with other ids
-      experimentName: experiment.name
+      experimentName: experiment.name,
+      variationName: variationNames.join(', '),
+      variationId: variationIds.join()
     };
 
-    // If this was a redirect experiment this value is made available
+    // If this was a redirect experiment and the effective referrer is different from document.referrer,
+    // this value is made available
     if (experiment.referrer) props.referrer = experiment.referrer;
 
-    // When only one variation running:
-    if (variations.length && variations.length < 2) {
-      props.variationName = variations[0].name;
-      props.variationId = variations[0].id;
-    }
-
     // When there is a multivariate experiment
-    if (variations.length > 1) {
-      props.variationName = variationNames.join(', '); // eg. 'Variation #1, Variation #2, Var 3709'
-      if (section) {
-        props.sectionName = section.name; // "grouping" of variations
-        props.variationId = section.variation_ids.join(); // eg '1,22,3709'
+    if (sections) {
+      // Since `sections` include all the possible sections on the page, we need to find the names of the sections
+      // if any of its variations were used. Experiments could display variations from multiple sections.
+      // So we have to do these terrible for loops as a workaround.
+      var sectionNames = [];
+      var sectionIds = [];
+      var sectionMap = {};
+      var activeSections = {};
+
+      // We need to reformat the data structure to reduce complexity from 4x to 2x
```

Comment:
This comment doesn't help me understand what data structure we're building in the following code or why -- I think it would be better to say something like
"Build a map from variationIds to sections so that we can quickly get the section name and id for any displayed variation."